test/e2e/aks_fleets_member.go

```diff
@@ -0,0 +1,165 @@
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservicefleet/armcontainerservicefleet"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AKSFleetsMemberInput struct {
+	Cluster       *clusterv1.Cluster
+	WaitIntervals []interface{}
+}
+
+const (
+	fleetName       = "capz-aks-fleets-manager"
+	updateGroupName = "capz-aks-fleets-member-update"
+)
+
+func AKSFleetsMemberSpec(ctx context.Context, inputGetter func() AKSFleetsMemberInput) {
+	input := inputGetter()
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	mgmtClient := bootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+
+	containerserviceClient, err := armcontainerservice.NewManagedClustersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	amcp := &infrav1.AzureManagedControlPlane{}
+	err = mgmtClient.Get(ctx, types.NamespacedName{
+		Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace,
+		Name:      input.Cluster.Spec.ControlPlaneRef.Name,
+	}, amcp)
+	Expect(err).NotTo(HaveOccurred())
+
+	groupClient, err := armresources.NewResourceGroupsClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("creating a resource group")
+	groupName := "capz-aks-fleets-member-" + amcp.Spec.ResourceGroupName
+	_, err = groupClient.CreateOrUpdate(ctx, groupName, armresources.ResourceGroup{
+		Location: ptr.To(os.Getenv(AzureLocation)),
+		Tags: map[string]*string{
+			"jobName":           ptr.To(os.Getenv(JobName)),
+			"creationTimestamp": ptr.To(os.Getenv(Timestamp)),
+		},
+	}, nil)
+	Expect(err).To(BeNil())
+
+	fleetClient, err := armcontainerservicefleet.NewFleetsClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	fleetsMemberClient, err := armcontainerservicefleet.NewFleetMembersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("creating a fleet manager")
+	poller, err := fleetClient.BeginCreateOrUpdate(ctx, groupName, fleetName, armcontainerservicefleet.Fleet{
+		Location: ptr.To(os.Getenv(AzureLocation)),
+	}, nil)
+	Expect(err).To(BeNil())
+	_, err = poller.PollUntilDone(ctx, nil)
+	Expect(err).To(BeNil())
+
+	By("Joining the cluster to the fleet hub")
+	var infraControlPlane = &infrav1.AzureManagedControlPlane{}
+	Eventually(func(g Gomega) {
+		aks, err := containerserviceClient.Get(ctx, amcp.Spec.ResourceGroupName, amcp.Name, nil)
+		g.Expect(err).NotTo(HaveOccurred())
+		Logf("AKS provisioning state -1: %s", *aks.ManagedCluster.Properties.ProvisioningState)
```

Comment:
Should we add a check to make sure `aks` is not nil before deref?

Reply:
Apologies, this is leftover from some debugging logs that I have removed.